src/mount/linux/mntent.rs

```diff
@@ -0,0 +1,136 @@
+//! Iterate over mtab/fstab
+
+use crate::{NixPath, Result};
+use libc::{endmntent, getmntent, mntent, setmntent, FILE};
+use std::ffi::{CStr, CString};
+
+#[derive(Debug)]
+/// A wrapper for `libc::mntent`, an iterator for `MountEntry`
+pub struct MountEntries {
+    file: *mut FILE,
+}
+
+impl MountEntries {
+    /// Creates a new `MountEntry iterator, opening the given mtab/fstab
+    ///
+    /// # Arguments
+    /// - `path` - Path to mtab/fstab, e.g. `/etc/mtab`.
+    /// - `mode` - Mode as for `fopen(3)`, e.g. `"r"` or `"a+"`.
+    ///
+    /// # See Also
+    /// [`getmntent`](https://www.man7.org/linux/man-pages/man3/getmntent.3.html)
+    /// [`fopen`](https://www.man7.org/linux/man-pages/man3/fopen.3.html)
+    pub fn new<P: ?Sized + NixPath>(path: &P, mode: String) -> Result<Self> {
+        let mode = CString::new(mode).unwrap();
+        let file = path.with_nix_path(|cstr| unsafe {
+            setmntent(cstr.as_ptr(), mode.as_ptr())
+        })?;
+
+        if file.is_null() {
+            panic!("The setmntent call returned NULL");
+        }
+
+        Ok(MountEntries { file })
+    }
+}
+
+impl Drop for MountEntries {
+    fn drop(&mut self) {
+        unsafe { endmntent(self.file) };
+    }
+}
+
+/// Represents an entry in mtab/fstab.
+#[derive(Debug, Eq, PartialEq)]
+pub struct MountEntry {
+    /// 1. name of the filesystem (e.g. the device)
+    pub fs_name: String,
```

Comment:
These are all owned strings. Is there a way we could do fancy life times to avoid copying?

Reply:
Possibly, they are only valid until the next call to the iteration, and I am not sure how to encode that so any pointers are welcome :)